### PR TITLE
Remove unused-result message when compiling

### PR DESCRIPTION
**Pull request by g-vidal**
warning messages about unused result occur.
This commit enables to remove the messages uding the option
-DWUNUSED=on
which makes a cleaner log

Signed-off-by: g-vidal <gerard.vidal@ens-lyon.fr>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ option (RPM "Generate RPM using CPack" OFF)
 option (NPM "Generate NPM/GYP tarballs" OFF)
 option (BUILDTESTS "Generate check-ups for upm" OFF)
 option (WERROR "Make all warnings into errors." ON)
+option (WUNUSED "Remove warnings if return value unused" OFF)
```
Comment (jtrulson) [inline CMakeLists.txt]: Maybe the text: "Enable compiler warnings regarding unused return values." 

would be clearer...?

File: CMakeLists.txt [hunk 2]
```diff
@@ -104,6 +105,13 @@ if (WERROR)
   message (STATUS "Warnings as errors enabled (-Werror), disable with -DWERROR=off")
 endif (WERROR)
 
+# Remove Warnings for "unused return"  ?
+if (WUNUSED)
+  list (APPEND C_CXX_WARNING_FLAGS -Wno-unused-result)
+   message (STATUS "Warnings for unused-result enabled, disable with -DWUNUSED=on")
```
Comment (jtrulson) [inline CMakeLists.txt]: The text: "...disable with -DWUNUSED=on"... Shouldn't that be "OFF", instead of "on"?

File: src/mhz16/mhz16.cxx [hunk 3]
```diff
@@ -37,12 +37,24 @@ MHZ16::MHZ16(int uart)
 {
```
Comment (jtrulson) [inline src/mhz16/mhz16.cxx]: These commits regarding mhz16 should be in a separate PR.

File: CMakeLists.txt (hunk 1 above)
Comment (jtrulson) [reply CMakeLists.txt]: I see - I was trying to remove the negative logic...  Here you set the default value to OFF.  Which means WUNUSED == OFF, and then the code below...

File: CMakeLists.txt (hunk 2 above)
Comment (jtrulson) [reply CMakeLists.txt]: ... checks to see if it's ON.  But here, if WUNUSED == ON, you tell them to set it to ON again to disable...  Hence the confusing logic...

What am I missing?

Comment (g-vidal) [reply CMakeLists.txt]: i'll make the message more explicit

File: src/mhz16/mhz16.cxx (hunk 3 above)
Comment (g-vidal) [reply src/mhz16/mhz16.cxx]: I think I have successfully rebased the other PR I will do hopefully the same for this one  soi'll close this one and open a new one .
You are right my mistake is to have used my dev branch for PRs I think the cleanest way is to cretae a temporary for each PR.
I'll cherrypick this commit in a separate branch

**Comment by pylbert** [comment]
@g-vidal, can you post a snippet of the unused warnings that you're encountering?

I suspect these are in the bindings (python, java, or javascript).  If so, maybe we can get away suppressing them there without adding another option to the top-level CMakeLists.txt?

**Comment by g-vidal** [comment]
You are right  @pylbert !
Most of the warnings come from the bindings  with few errors repeated again and again. Towards the end of the compilation I get huge messages. Thiese are examples from the beginning.
I have made a new cleaner PR after comments from @jtrulson #642.

/home/vidal/Logiciels/IoT/upm/build/src/interfaces/interfacesJAVASCRIPT_wrap.cxx: In function ‘void SWIGV8_AddStaticVariable(v8::Handle<v8::Object>, const char*, SwigV8AccessorGetterCallback, SwigV8AccessorSetterCallback)’:
/home/vidal/Logiciels/IoT/upm/build/src/interfaces/interfacesJAVASCRIPT_wrap.cxx:1505:87: warning: ignoring return value of ‘v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute)’, declared with attribute warn_unused_result [-Wunused-result]
   obj->SetAccessor(SWIGV8_CURRENT_CONTEXT(), SWIGV8_SYMBOL_NEW(symbol), getter, setter);
                                                                                       ^
In file included from /usr/local/include/node/node.h:63:0,
                 from /home/vidal/Logiciels/IoT/upm/build/src/interfaces/interfacesJAVASCRIPT_wrap.cxx:171:
/usr/local/include/node/v8.h:3240:37: note: declared here
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(Local<Context> context,


/home/vidal/Logiciels/IoT/upm/build/src/grovescam/grovescamJAVASCRIPT_wrap.cxx: In function ‘void SWIGV8_AddStaticVariable(v8::Handle<v8::Object>, const char*, SwigV8AccessorGetterCallback, SwigV8AccessorSetterCallback)’:
/home/vidal/Logiciels/IoT/upm/build/src/grovescam/grovescamJAVASCRIPT_wrap.cxx:1505:87: warning: ignoring return value of ‘v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute)’, declared with attribute warn_unused_result [-Wunused-result]
   obj->SetAccessor(SWIGV8_CURRENT_CONTEXT(), SWIGV8_SYMBOL_NEW(symbol), getter, setter);
                                                                                       ^
In file included from /usr/local/include/node/node.h:63:0,
                 from /home/vidal/Logiciels/IoT/upm/build/src/grovescam/grovescamJAVASCRIPT_wrap.cxx:171:
/usr/local/include/node/v8.h:3240:37: note: declared here
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(Local<Context> context,
                                     ^~~~~~~~~~~

**Comment by pylbert** [comment]
@g-vidal, this helps, however are there two different issues here?

1. Suppressing the warnings added from the V8_WARN_UNUSED_RESULT macro
2. An ABI change in GCC 7.1 for arm.  

If we can, I would like to suppress the unused results in the Javascript wrappers by adding -Wno-unused-result at the SWIG-level.

https://github.com/intel-iot-devkit/upm/blob/306872619a413bfbfaced453e11ab024237684d2/src/CMakeLists.txt#L6-L10

For those other notes about the change in 7.1 - I wouldn't imagine the -Wno-unused-result gets rid of these.  Have you found a way around them?
